lib/agent0/agent0/accounts_config.py

```diff
@@ -0,0 +1,190 @@
+        return env_str
+
+
+def initialize_accounts(
+    agent_config: list[AgentConfig], env_file: str | None = None, random_seed: int = 1, develop: bool = False
+) -> AccountKeyConfig:
+    """
+    Build or loads an accounts environment file
+    If it doesn't exist, create it based on agent_config
+    (if develop is off, print instructions on adding in user private key and running script to fund bots).
+    If it does exist, read it in and use it
+    """
+    # Default location
+    if env_file is None:
+        env_file = "account.env"
+
+    if not os.path.exists(env_file):
+        logging.info("Creating %s", env_file)
+        # Create AccountKeyConfig from agent config
+        account_key_config = build_account_key_config_from_agent_config(agent_config, random_seed)
+        # Create file
+        with open(env_file, "w", encoding="UTF-8") as file:
+            file.write(account_key_config.to_env_str())
+        if not develop:
+            print(
+                f"Account key config written {env_file}. "
+                "Run the following command to fund the accounts, then rerun this script."
+            )
+            # Different commands depending on if default env file is used
+            if env_file == "account.env":
+                print("python lib/agent0/bin/fund_bots_from_user_key.py -u <user_private_key>")
+            else:
+                print(f"python lib/agent0/bin/fund_bots_from_user_key.py -u <user_private_key> -f {env_file}")
+            # Clean exit
+            sys.exit(0)
+    else:
+        logging.info("Loading %s", env_file)
+        # Ensure account_config matches up with env_file
+        # TODO this is where we would read state of bot from chain
+        account_key_config = build_account_config_from_env(env_file)
+        num_total_bots = sum(agent.number_of_agents for agent in agent_config)
+        if num_total_bots != len(account_key_config.AGENT_KEYS):
+            raise ValueError("Number of bots in agent config does not match number of bots in env file")
+
+    return account_key_config
+
+
+def build_account_key_config_from_agent_config(
+    agent_configs: list[AgentConfig], random_seed: int, user_key: str | None = None
+) -> AccountKeyConfig:
+    """Build an Account Config from a provided agent config.
+
+    Arguments
+    --------
+    user_key: str
+        The provided user key to use
+    agent_configs: list[AgentConfig]
+        The provided agent configs
+
+    Returns
+    -------
+    AccountConfig
+        Config settings required to connect to the eth node
+    """
+    rng = np.random.default_rng(random_seed)
+    agent_private_keys = []
+    agent_base_budgets = []
+    agent_eth_budgets = []
+    for agent_info in agent_configs:
+        for _ in range(agent_info.number_of_agents):
+            agent_private_keys.append(make_private_key())
+
+            if isinstance(agent_info.eth_budget, Budget):
```

Comment:
This cleans up nicely with [structured pattern matching](https://peps.python.org/pep-0622/#rationale-and-goals).
```
match agent_info.base_budget:
    case Budget as sampler_budget: # sample if a Budget object
        kwargs["budget"] = sampler_budget.sample_budget(rng) # TODO: rename this function to `sample()`
    case FixedPoint as scalar_budget: # sample is a fixed-point
        kwargs["budget"] = scalar_budget.scaled_value # TODO: rename this member variable to `fixed_value`
    case _ as budget_type:
        raise ValueError(f"Invalid base_budget type: {budget_type)}")
```